sockeye/inference.py

```diff
@@ -802,42 +848,51 @@ def _beam_search(self,
         # Maximum output length
         max_output_length = self.models[0].get_max_output_length(source_length)
 
-        # sequences: (beam_size, output_length), pre-filled with <s> symbols on index 0
-        sequences = mx.nd.array(np.full((self.beam_size, max_output_length), C.PAD_ID), dtype='int32', ctx=self.context)
+        # General data structure: each row has batch_size * beam blocks for the 1st sentence, with a full beam,
+        # then the next block for the 2nd sentence and so on
+
+        # sequences: (batch_size * beam_size, output_length), pre-filled with <s> symbols on index 0
+        sequences = mx.nd.full((self.batch_size * self.beam_size, max_output_length), val=C.PAD_ID, ctx=self.context, dtype='int32')
         sequences[:, 0] = self.start_id
 
-        lengths = mx.nd.ones((self.beam_size, 1), ctx=self.context)
-        finished = mx.nd.zeros((self.beam_size,), dtype='int32', ctx=self.context)
+        lengths = mx.nd.ones((self.batch_size * self.beam_size, 1), ctx=self.context)
+        finished = mx.nd.zeros((self.batch_size * self.beam_size,), ctx=self.context, dtype='int32')
 
         # attentions: (beam_size, output_length, encoded_source_length)
```

Comment:
the comment about the dimensionality is not up to date.

Reply:
fixed